polaris-react/src/components/Text/Text.tsx

```diff
@@ -72,7 +72,7 @@ export const Text = ({
     styles.root,
     styles[variant],
     fontWeight ? styles[fontWeight] : styles[VariantFontWeightMapping[variant]],
-    (alignment || truncate) && styles.block,
+    (alignment || truncate) && as !== 'span' && styles.block,
```

Comment:
My gut feeling is we shouldn't set an alignment ever. `h1` is `block` and `span` is `inline` by default unless there is a reset doing some magic.

Reply:
Fair point. So if someone set alignment on a `span` then the expected behaviour would be nothing.

Making alignment switch the element to block was based on a previous iteration when all the `Display` elements were `span`, but that's no longer the case so we should be good to remove that.